api/src/main/java/com/redhat/ipaas/api/IPaasEntity.java

```diff
@@ -15,9 +15,22 @@
  */
 package com.redhat.ipaas.api;
 
-public interface IPaasEntity {
+public abstract class IPaasEntity {
 
-	public String getId();
-	public void setId(String id);
+	abstract public String getId();
+	abstract public void setId(String id);
+	
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
It also makes testing harder as we can't rely on equals to test deep equality so will have to test individual field equality which sucks.